Sources/Navidux/Core/Navigation/NavigationAction/Actions/PushNavigationAction.swift

```diff
@@ -0,0 +1,39 @@
+//
+//  PushNavigationAction.swift
+//
+//
+//  Created by Tamerlan Satualdypov on 13.02.2024.
+//
+
+import Foundation
+
+public struct PushNavigationAction: NavigationAction {
```

Comment:
`push` semantics is not working here. it's about working with stack nad navigation controller is some kind of stack. need some other pair of vebrs not push/pop

@darkkifir @baluhman @Bananzaaa need your opinon too

Reply:
I don't understand why push semantics does not work, because we still work with navigation stack underneath and it would be less confusing to keep push/pop naming for actions

Reply:
in this same file in perform function:
```swift
        switch self.presentationStyle {
        case .fullScreen:
            coordinator.navigationController.pushViewController(screen, animated: true)
        case .modal:
            coordinator.navigationController.present(screen, animated: true, completion: nil)
        }
```

in `>Push<NavigationAction` you have `>present<Style` which leads to `push` in navigation controller or `present` above it

Reply:
The idea is that `present` function will still add screen to the stack, so all screens are being held and manipulated in the stack. It is not implemented yet, as I have not started refactoring `NavigationController`